pkgs/collection/lib/src/equality.dart

```diff
@@ -325,16 +325,19 @@ class MapEquality<K, V> implements Equality<Map<K, V>> {
     var length = map1.length;
     if (length != map2.length) return false;
     Map<_MapEntry, int> equalElementCounts = HashMap();
+    var values1 = map1.values.iterator;
     for (var key in map1.keys) {
-      var entry = _MapEntry(this, key, map1[key]);
-      var count = equalElementCounts[entry] ?? 0;
-      equalElementCounts[entry] = count + 1;
+      var value = (values1..moveNext()).current;
+      var entry = _MapEntry(this, key, value);
+      equalElementCounts.update(entry, (i) => i + 1, ifAbsent: () => 1);
     }
+    final values2 = map2.values.iterator;
     for (var key in map2.keys) {
-      var entry = _MapEntry(this, key, map2[key]);
-      var count = equalElementCounts[entry];
-      if (count == null || count == 0) return false;
-      equalElementCounts[entry] = count - 1;
+      var value = (values2..moveNext()).current;
+      var entry = _MapEntry(this, key, value);
+      var count =
+          equalElementCounts.update(entry, (i) => i - 1, ifAbsent: () => -1);
```

Comment:
Can we define these callbacks statically?

Reply:
Done